packages/adblocker-webextension/adblocker.ts

```diff
@@ -560,9 +560,9 @@ export class WebExtensionBlocker extends FiltersEngine {
     browser: Browser,
     msg: IBackgroundCallback & { action?: string },
     sender: Runtime.MessageSender,
-  ): Promise<IMessageFromBackground | undefined> => {
+  ): Promise<Partial<IMessageFromBackground> | undefined> => {
     return new Promise((resolve, reject) => {
-      this.handleRuntimeMessage(browser, msg, sender, resolve).catch(reject);
+      this.handleRuntimeMessage(browser, msg, sender, resolve).catch(reject).finally(() => resolve({}));
```

Comment:
Won’t this always return an empty object?

Reply:
there is no such risk as `handleRuntimeMessage` has a await at its end https://github.com/ghostery/adblocker/blob/64b2e0859e3f5c35e0a61fc4d0b209a442d4165f/packages/adblocker-webextension/adblocker.ts#L518